linera-base/src/data_types.rs

```diff
@@ -845,9 +846,11 @@ pub struct ApplicationDescription {
 
 impl From<&ApplicationDescription> for ApplicationId {
     fn from(description: &ApplicationDescription) -> Self {
-        ApplicationId::new(CryptoHash::new(&BlobContent::new_application_description(
-            description,
-        )))
+        let mut hash = CryptoHash::new(&BlobContent::new_application_description(description));
+        if matches!(description.module_id.vm_runtime, VmRuntime::Evm) {
+            hash.set_as_evm();
```

Comment:
```suggestion
            hash.make_evm_compatible();
```

Reply:
Yes, of course. You do not have to precise this as it is implied by your previous change.